src/KindPage.tsx

```diff
@@ -209,7 +214,8 @@ function KindPage({
     setCreateEntityDialogIsOpen(false);
   }, []);
 
-  useDocumentTitle(kind);
+  const decodedKind = decodeURIComponent(kind);
```

Comment:
Can you move the decodeURIComponent to the call site (in DatastoreAdmin?). Then the changes in this component shouldn't be necessary.